### DISCUSSION TITLE
Couldn't see the changes in localhost

### DISCUSSION THREAD

**Discussion by skwowet** [supabase/supabase#3719] [Questions] [answered]
Hi there,

I have been working on 2 bug fixes for the website and I followed [instructions](https://github.com/supabase/supabase/tree/master/www) but couldn't see the changes in localhost. am I doing something wrong? while setting up the localhost?

Any help regarding this would be appreciated, thanks in advance!

**Comment by dshukertjr** [comment]
Hi @yeganathan18!

If you could explain the steps you have taken step by step as well as which file you modified, it'll make it easier to assist you!

**Reply by skwowet** [reply]
So first I cloned the [supabase](https://github.com/supabase/supabase) repo and did the following in order to run the site locally in localhost,

1. Opened the terminal in the parent directory and ran the following commands, in three separate tabs as instructed in the instructions of the supabase [www directory](https://github.com/supabase/supabase/tree/master/www)
```
# tab 1
make dev

# tab 2
cd www
npm install
npm run dev

# tab 3
cd web/spec
make
npm install
npm run gen:supabase
npm run start
```
2. The site started running in the localhost:3000 and I started fixing the bugs but wasn't able to view the changes in localhost.
3. I tried restarting it but still nothing happens.

Hold on, there was an error while executing the commands in tab 3 part, the error occurred while I was executing commands in the supabase/web/spec directory
<img width="747" alt="Screenshot 2021-10-30 at 9 23 58 PM" src="https://user-images.githubusercontent.com/63534555/139540318-becae838-5491-4188-a457-15dcea0d9a99.png">

Ps: I don't think the error that I mentioned causes the problem 🤔

**Reply by dshukertjr** [reply]
@yeganathan18 Thanks for sharing how you got started. Could you also share which file you modified during your bug fix as well?

**Reply by MildTomato** [reply]
try running `npm run start` in the folder /web, not /web/spec

**Reply by skwowet** [reply]
@dshukertjr in order to check whether I'm getting output in the localhost, I tried to change this file - www/components/CTABanner/index.tsx

Just the text from **Build in a weekend** to **Hello World!**

**Reply by skwowet** [reply]
@MildTomato I'm getting the same error as stated above (screenshot), even if I run the npm command in the web directory :(

**Reply by MildTomato** [reply]
it looks like `npm run gen:supabase` wasn't run, i think thats what would generate those reference files

**Reply by MildTomato** [reply]
if you're working on the www site though, you just need to `cd www` and run `npm run dev`.
the `web` folder is the docs website

**Reply by MildTomato** [reply]
these instructions are out of date actually - you no longer need to run `make dev` in the root of supabase.

you can just run `www` nextjs site and the `web` docusaurus sites separately

**Reply by Isaiah-Hamilton** [reply]
Try this if you still need help
1. start from `/supabase`
2. Then `cd /www`
3. Then `npm install` or `yarn install`
4. Then `npm run dev` or `yarn dev`
5. Then go to your localhost on port 3000

This is how I start my localhost to test my code.

**Reply by skwowet** [reply]
@MildTomato I just need to run the `www` nextjs site, in order to fix a bug and followed instructions as you said, but **404 Page was not found** was the result in localhost:3000 ☹️

browser:
<img width="1792" alt="Screenshot 2021-11-03 at 8 23 02 PM" src="https://user-images.githubusercontent.com/63534555/140084577-a6cdf08e-33a2-4240-a578-e09e54084b4c.png">

terminal:
<img width="1445" alt="Screenshot 2021-11-03 at 8 23 24 PM" src="https://user-images.githubusercontent.com/63534555/140084746-5747c414-ca0a-443a-a1ff-c0df4999cc73.png">

Ps: Also, checked by running `npm run build` but same 404 error.

**Reply by Isaiah-Hamilton** [reply]
@yeganathan18 You need to open you local host with `localhost:3000/new`

You need to add `/new` to the end of your link because of the way vercel and next is configured. Let me now if that helped.

**Reply by skwowet** [reply]
@Isaiah-Hamilton GG! Everything set 🎉

**Reply by MildTomato** [reply]
> @yeganathan18 You need to open you local host with localhost:3000/new

this is out of date. with the latest changes in `master` branch, the /new url is no longer a requirement

**Comment by Isaiah-Hamilton** [comment, marked as answer]
This discussions was answered with this [answer](https://github.com/supabase/supabase/discussions/3719#discussioncomment-1582613).

1. start from `/supabase`
2. Then `cd /www`
3. Then `npm install` or `yarn install`
4. Then `npm run dev` or `yarn dev`
5. Then go to `localhost:3000`, and you should see your changes.
